base/dates/io.jl

```diff
@@ -324,22 +450,48 @@ generate the string "1996-01-15T00:00:00" you could use `format`: "yyyy-mm-ddTHH
 Note that if you need to use a code character as a literal you can use the escape character
 backslash. The string "1996y01m" can be produced with the format "yyyy\\ymm\\m".
 """
-format(dt::TimeType,f::AbstractString;locale::AbstractString="english") = format(dt,DateFormat(f,locale))
+format(dt::TimeType,f::AbstractString;locale=ENGLISH) = format(dt,DateFormat(f,locale))
+
+# show
+
+function Base.show(io::IO, dt::DateTime)
+    if millisecond(dt) == 0
+        format(io, dt, dateformat"YYYY-mm-dd\THH:MM:SS")
+    else
+        format(io, dt, dateformat"YYYY-mm-dd\THH:MM:SS.s")
+    end
+end
+
+function Base.show(io::IO, dt::Date)
+    format(io, dt, dateformat"YYYY-mm-dd")
+end
+
+function Base.string(dt::DateTime)
+    if millisecond(dt) == 0
+        format(dt, dateformat"YYYY-mm-dd\THH:MM:SS", 24)
+    else
+        format(dt, dateformat"YYYY-mm-dd\THH:MM:SS.s", 26)
```

Comment:
No, `s` stands for any number of digits e.g. `s` gives `.5` or `.05` or `.005` while `.sss` would give `.500`, `.050`.